pjlib/src/pj/os_core_unix.c

```diff
@@ -315,7 +315,7 @@ PJ_DEF(pj_bool_t) pj_thread_is_registered(void)
  * A cool solution would be to port (if possible) the code from the
  * android os regarding set_sched groups.
  */
-#if PJ_ANDROID
+#if defined(PJ_JNI_HAS_JNI_ONLOAD) && PJ_JNI_HAS_JNI_ONLOAD != 0
```

Comment:
- The function `set_android_thread_priority()` defined below should be Android specific, also in `pj_thread_set_prio()` below, it is called under `PJ_ANDROID` protection, so it needs `PJ_ANDROID` protection. But then it also calls `pj_jni_attach_jvm()` which needs `PJ_JNI_HAS_JNI_ONLOAD` protection. So perhaps it needs both protections?
- Perhaps we need to make sure any other parts in the lib invoking `pj_jni_attach_jvm()` (if any, the `PJ_DEF` and the naming indicates it is a public API) are protected by `PJ_JNI_HAS_JNI_ONLOAD` too, instead of `PJ_ANDROID`. Or alternatively, define a dummy/empty function version when `PJ_JNI_HAS_JNI_ONLOAD` is set to zero.

Reply:
Done. I have provided alternative/fallback implementation for thread priority setting and guid. Now it can build and link successfully.

The following components have to be specifically disabled (refer to the updated PR desc):
Android JNI, Oboe, and android_dev
The question is should we automatically disable them (by adding `PJ_JNI_HAS_JNI_ONLOAD` protection) or leave it to the user to disable them?
I prefer the latter option so that user is aware that those components won't be available.

Reply:
So it seems that `PJ_JNI_HAS_JNI_ONLOAD` is currently used by Android features only (while it sounds like a generic JNI thing), perhaps that's why it is set to `PJ_ANDROID`.

IMO the only reason `PJ_JNI_HAS_JNI_ONLOAD` not set is that there is already `JNI_OnLoad` implemented somewhere else where PJSIP is integrated to it as a single dynamic library. In this case, perhaps PJSIP should provide a public API to set its JavaVM/`pj_jni_jvm` as alternative to getting JavaVM via `JNI_OnLoad` (note that there can only be one `JNI_OnLoad` in a library loaded from Java).